### ISSUE TITLE
Severe performance degradation between 2.3.x and 2.4.0 stable

### ISSUE THREAD

**Issue by StarLightPL** [docker/for-win#9037]
- [x] I have tried with the latest version of my channel (Stable or Edge)
  - [ ] I have uploaded Diagnostics

### Expected behavior
No noticeable difference in operation between 2.3.x and 2.4.x

### Actual behavior
On version 2.3.4.0 stable, startup of all my containers took 3m 30s, after upgrading to 2.4.0 startup takes 5m 30m. This behaviour is entirely reversible by downgrading to previous major version of Docker Desktop (2.3)

### Information
Issue is easily reproducible by switching Docker Desktop version only, without touching anything else. It doesn't matter if containers are pruned and rebuilt for particular version, "normal" run time (using prebuilt cache) stays the same.
Docker Desktop is running on native windows instance.
Problem appeared with 2.4.0 stable update.
Windows version 2004 (19041.508), latest Ubuntu on WSL2.
Using docker network bridge between containers.
Using ansible (ansible_connection=docker) to run some actions inside the containers during aforementioned startup.

### Steps to reproduce the behavior
Startup times differ predictively between aforementioned versions.

**Comment by stephen-turner** [comment]
Were you running on the WSL 2 backend in 2.3, or were you on the Hyper-V backend then?

**Comment by StarLightPL** [comment]
I'm pretty sure it was WSL2 backend in both instances. I will reinstall and check though, I will also be able to check if the ansible part is operating slower or not.

**Comment by StarLightPL** [comment]
OK I have the results - it's ansible operating on docker containers using connection=docker that's causing the massive slowdown.

Testing "methodology":
I downgraded to 2.3.0.4 (46911), rebuilt my containers and then made two runs (basically running the containers, then running some ansible on them), back to back. I recorded logfiles from each run, then measured the startup times of each container and ansible operation time. Then I upgraded (using Docker Desktop's prompt) to current stable 2.4.0.0 (48506) and repeated the procedure.

**Ansible 2.9.2 is almost twice as slow in newer docker desktop version** (2m24s vs 4m09s on run number 1, and 2m22s vs 4m11s on run number 2).  So it seems that bug is borderline docker desktop and ansible, great... Like mentioned, reverting to 2.3.4 helps.

There are some slight differences on running the containers, but they seem negligible and prone to measurement error because they are within a second or two difference.
